server/swap/swap.go

```diff
@@ -0,0 +1,1171 @@
+	return &waitSettings{
+		// must smarten up this expiration value before merge. Where should this
+		// come from?
+		expiration: time.Now().Add(txWaitExpiration),
+		accountID:  user,
+		request:    req,
+		timeoutErr: &rpc.RPCError{
+			Code:    rpc.TransactionUndiscovered,
+			Message: fmt.Sprintf("failed to find transaction %s", txid),
+		},
+	}
+}
+
+// messageAcker is information needed to process the user's
+// rpc.AcknowledgementResult.
+type messageAcker struct {
+	user    account.AccountID
+	match   *matchTracker
+	params  rpc.Signable
+	isMaker bool
+	isAudit bool
+}
+
+// processAck processes a single rpc.AcknowledgementResult, validating the
+// signature and updating the (order.Match).Sigs record.
+func (s *Swapper) processAck(resp *rpc.Response, acker *messageAcker) {
+	if resp.ID == nil || !rpc.IsValidIDType(*resp.ID) {
+		s.respondError(-1, acker.user, rpc.IDTypeError,
+			fmt.Sprintf("unsupported ID type: %T", resp.ID))
+		return
+	}
+	var ack rpc.AcknowledgementResult
+	err := json.Unmarshal(resp.Result, &ack)
+	if err != nil {
+		s.respondError(*resp.ID, acker.user, rpc.RPCParseError,
+			fmt.Sprintf("error parsing audit notification acknowledgment: %v", err))
+		return
+	}
+	// Check the signature.
+	sigBytes, err := hex.DecodeString(ack.Sig)
+	if err != nil {
+		s.respondError(*resp.ID, acker.user, rpc.SignatureError,
+			fmt.Sprintf("error decoding signature %s: %v", ack.Sig, err))
+		return
+	}
+	msg, err := acker.params.Serialize()
+	if err != nil {
+		s.respondError(*resp.ID, acker.user, rpc.SerializationError,
+			fmt.Sprintf("unable to serialize mathch params: %v", err))
+		return
+	}
+	err = s.authMgr.Auth(acker.user, msg, sigBytes)
+	if err != nil {
+		s.respondError(*resp.ID, acker.user, rpc.SignatureError,
+			fmt.Sprintf("signature validation error: %v", err))
+		return
+	}
+	// Set the appropriate signature, based on the current step and actor.
+	// DRAFT NOTE: Using the matchMtx like this is not ideal. It might be better
+	// to have individual locks on the matchTrackers.
+	s.matchMtx.Lock()
+	defer s.matchMtx.Unlock()
+	if acker.isAudit {
+		if acker.isMaker {
+			acker.match.Sigs.MakerAudit = sigBytes
+		} else {
+			acker.match.Sigs.TakerAudit = sigBytes
+		}
+	} else {
+		if acker.isMaker {
+			acker.match.Sigs.MakerRedeem = sigBytes
+		} else {
+			acker.match.Sigs.TakerRedeem = sigBytes
+		}
+	}
+}
+
+// processInit processes the `init` RPC request, which is used to inform the
+// DEX of a newly broadcast swap transaction. Once the transaction is seen and
+// and audited by the Swapper, the counter-party is informed with an 'audit'
+// request. This method is run as a chainWaiter.
+func (s *Swapper) processInit(req *rpc.Request, params *rpc.InitParams, stepInfo *stepInformation) bool {
+	if !rpc.IsValidIDType(req.ID) {
+		s.respondError(-1, stepInfo.actor.user, rpc.IDTypeError,
+			fmt.Sprintf("unsupported ID type: %T", req.ID))
+		return dontTryAgain
+	}
+	// Validate the swap contract
```

Comment:
Note that as of right now, there is no check to ensure that the maker's swap transaction has `SwapConf` confirmations before the taker broadcasts their swap.